### ISSUE TITLE
Initializing TradingAlgorithm Object Does Not Set _analyze Method in algorithm.py

### ISSUE THREAD

**Issue by elmq0022** [quantopian/zipline#801]
The following code does not call the passed analyze function when trying to work directly/manually with zipline (i.e. not from ipython or command line)

```python
import pytz
from zipline.api import order, record, symbol, history, add_history, order_target
from datetime import datetime
from zipline.algorithm import TradingAlgorithm
from zipline.utils.factory import load_bars_from_yahoo

start = datetime(2000, 1, 1, 0, 0, 0, 0, pytz.utc)
end = datetime(2001, 1, 1, 0, 0, 0, 0, pytz.utc)
data = load_bars_from_yahoo(stocks=['AAPL'],start=start, end=end)

def initialize(context):
   pass

def handle_data(context, data):
    order(symbol('AAPL'), 10)
    record(APPL=data[symbol('AAPL')].price)

def analyze(context, perf):
   print(perf.head())
    
algo_obj = TradingAlgorithm(initialize=initialize, handle_data=handle_data, analyze=analyze)
perf= algo_obj.run(data)
```

I think it can be fixed by updating the algorithm.py file with the last line of the following code. 

```python
elif kwargs.get('initialize', False) and kwargs.get('handle_data'):
            if self.algoscript is not None:
                raise ValueError('You can not set script and \
                initialize/handle_data.')
            self._initialize = kwargs.pop('initialize')
            self._handle_data = kwargs.pop('handle_data')
            self._analyze = kwargs.pop('analyze', None)
```

I'd update the code and submit a pull request, but I'm not 100% if it's an issue.

**Comment by richafrank** [comment]
@elmq0022 That fix makes sense to me - want to open a PR?

I suggest adding that new `analyze` line after we set `self._before_trading_start`, just for consistency with the previous `if` block.

**Comment by elmq0022** [comment]
@richafrank Thanks for the feedback and the suggested placement for consistency.  I'll submit a pull request sometime this weekend.

**Comment by richafrank** [comment]
Sounds great!

**Comment by richafrank** [comment]
Fixed with #819.
